torch/lib/THD/base/channels/ChannelUtils.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <cstdlib>
+#include <cstdint>
+#include <string>
+#include <system_error>
+#include <tuple>
+
+namespace thd {
+
+#define SYSCHECK(expr) { \
+  errno = 0; (expr);     \
+  if (errno != 0) throw std::system_error(errno, std::system_category()); \
+}
+
+template<typename T>
+void send_bytes(int socket, const T* buffer, std::size_t length)
```

Comment:
Why did you remove that `MSG_MORE` optimization? Either keep it, or even better add a specialization for vector writes.